tests/compute/dynamic-dispatch-14.slang

```diff
@@ -29,8 +29,7 @@ RWStructuredBuffer<int> gOutputBuffer;
 //TEST_INPUT: set gCb = new StructuredBuffer<IInterface>{new MyImpl{1}};
 RWStructuredBuffer<IInterface> gCb;
 
-// Add two elements into the structured buffer to prevent specialization.
-//TEST_INPUT: set gCb1 = new StructuredBuffer<IInterface>{new MyImpl{1}, new MyImpl2{2}};
+//TEST_INPUT: set gCb1 = new StructuredBuffer<IInterface>{new MyImpl{1}} : specialization_args(__Dynamic);
```

Comment:
Given the way that the new `TEST_INPUT` stuff uses a prefix `new` for establishing a sub-object based on the following expression, it might make sense to have the specialization-argument information work the same way. If we have:

```
struct MyStuff
{
    int a;
    IThing thing;
}
ConstantBuffer<MyStruct> cb;
```

we might use:

```
set cb = new { 99, dynamic new ConcreteThing{ ... } };
```

Here the prefix operator `dynamic` wraps a value and when binding to a `ShaderOffset` it first calls `setSpecializationArg()` for that offset to set up dynamic dispatch, and then recursively sets the wrapped value into the same offset.

A similar prefix operator could be used for explicit static specialization:

```
set cb = new { 99, specialize OtherThing new ConcreteThing{ ... } };
```

The above would say "specialize the code for `OtherThing`, but I'm actually binding a `ConcreteThing`, so I guess that's undefined behavior..."